OpenRA.Mods.RA/Widgets/ColorMixerWidget.cs

```diff
@@ -95,28 +95,25 @@ void GenerateBitmapWorker()
 					hue = H;
 				}
 
-				lock (backBitmap)
+				lock (back)
 				{
-					var bitmapData = backBitmap.LockBits(backBitmap.Bounds(),
-						ImageLockMode.ReadWrite, PixelFormat.Format32bppArgb);
-
 					unsafe
 					{
-						int* c = (int*)bitmapData.Scan0;
-
 						// Generate palette in HSV
-						for (var v = 0; v < 256; v++)
-							for (var s = 0; s < 256; s++)
-								*(c + (v * bitmapData.Stride >> 2) + s) = HSLColor.FromHSV(hue, s / 255f, (255 - v) / 255f).RGB.ToArgb();
+						fixed (byte* cc = &back[0])
+						{
+							var c = (int*)cc;
+							for (var v = 0; v < 256; v++)
+								for (var s = 0; s < 256; s++)
+									*(c + (v * 256) + s) = HSLColor.FromHSV(hue, s / 255f, (255 - v) / 255f).RGB.ToArgb();
+						}
 					}
 
-					backBitmap.UnlockBits(bitmapData);
-
-					lock (frontBitmap)
+					lock (front)
```

Comment:
https://scan8.coverity.com/reports.htm#v24414/p10888/fileInstanceId=5945238&defectInstanceId=2281120&mergedDefectId=111052 exposes the threading issue. A bad choice of lock objects. However I don't know enough to fix this properly myself. Coverity suggests

> Instead of using this.front as a lock, create a readonly field of type Object which is only used as a lock.

Not sure how you lock/unlock with a readonly field.